src/main/java/io/openliberty/tools/intellij/actions/LibertyProjectAction.java

```diff
@@ -46,13 +45,19 @@ public void actionPerformed(@NotNull AnActionEvent e) {
         }
         List<BuildFile> buildFileList = getBuildFileList(project);
         if (!buildFileList.isEmpty()) {
-            final String[] projectNames = buildFileToProjectNames(buildFileList);
-            final int ret = Messages.showChooseDialog(project,
+            String[] projectName = extractBuildFileAttributes(buildFileList, BuildFile::getProjectName);
+            String[] projectPath = extractBuildFileAttributes(buildFileList, buildFile -> buildFile.getBuildFile().getPath());
```

Comment:
I'm not suggesting any changes here but noticed this is looping over the same list twice. Seems like we're gathering data into parallel arrays that could have been aggregated. Again, not suggesting you change this. Just something to think about when performance matters.